taui/src/components/route-card.js

```diff
@@ -14,6 +14,61 @@ type Props = {
 }
 
 export default class RouteCard extends React.PureComponent<Props> {
+  constructor (props) {
+    super(props)
+    this.summaryImage = this.summaryImage.bind(this)
+  }
+
+  summaryImage (props) {
+    const nprops = props.nprops
+
+    // Look for an available image to use as the summary
+    let imageField = nprops['town_square_thumbnail'] ? 'town_square' : null
+    if (!imageField) {
+      imageField = nprops['open_space_or_landmark_thumbnail'] ? 'open_space_or_landmark' : null
+    }
+    if (!imageField) {
+      imageField = nprops['school_thumbnail'] ? 'school' : null
+    }
+    if (!imageField) {
+      imageField = 'street'
+    }
+
+    if (!imageField) {
+      return null // Have no summary image available
+    }
+
+    const description = nprops[imageField + '_description']
+    const license = nprops[imageField + '_license']
+    const licenseUrl = nprops[imageField + '_license_url']
+    const imageLink = nprops[imageField]
+    const thumbnail = nprops[imageField + '_thumbnail']
+    const userName = nprops[imageField + '_username']
+
+    if (!thumbnail) {
+      return null
+    }
+
+    // Build the attribution text to display on hover
+    let attrText = userName + ' [' + license
```

Comment:
It might be helpful to create an `AttributedImage` component that you could plug in both here and in the neighborhood details above; that would let you avoid duplicating this code to generate the attribution. Feel free to disregard if there are obstacles to that that I'm not seeing, or if it would be significant extra work.